scripts/collection_from_items.py

```diff
@@ -6,19 +6,35 @@
 from boto3 import client
 from linz_logger import get_log
 
-from scripts.cli.cli_helper import coalesce_multi_single
+from scripts.cli.cli_helper import coalesce_multi_single, valid_date
 from scripts.files.fs_s3 import bucket_name_from_path, get_object_parallel_multithreading, list_json_in_uri
 from scripts.logging.time_helper import time_in_ms
 from scripts.stac.imagery.collection import ImageryCollection
+from scripts.stac.imagery.generate_metadata import generate_description, generate_title
 from scripts.stac.imagery.provider import Provider, ProviderRole
 
 
+# pylint: disable-msg=too-many-locals
 def main() -> None:
     parser = argparse.ArgumentParser()
+    parser.add_argument("--preset", dest="preset", required=True, help="Standardised file format. Example: webp")
     parser.add_argument("--uri", dest="uri", help="s3 path to items and collection.json write location", required=True)
     parser.add_argument("--collection-id", dest="collection_id", help="Collection ID", required=True)
-    parser.add_argument("--title", dest="title", help="Collection title", required=True)
-    parser.add_argument("--description", dest="description", help="Collection description", required=True)
+    parser.add_argument("--subtype", dest="subtype", help="Dataset subtype description", required=True)
+    parser.add_argument("--region", dest="region", help="Region of Dataset", required=True)
+    parser.add_argument("--gsd", dest="gsd", help="GSD of imagery Dataset", required=True)
+    parser.add_argument("--location", dest="location", help="Optional Location of dataset, e.g.- Hutt City", required=False)
+    parser.add_argument(
+        "--start-datetime", dest="start_datetime", help="Start datetime in format YYYY-MM-DD", type=valid_date, required=True
+    )
+    parser.add_argument(
+        "--end-datetime", dest="end_datetime", help="End datetime in format YYYY-MM-DD", type=valid_date, required=True
+    )
+    parser.add_argument("--event", dest="dest", help="Event name if applicable", required=False)
+    parser.add_argument(
+        "--historic-survey-number", dest="historic_survey_number", help="Historic Survey Number if Applicable", required=False
```

Comment:
I have added an example but the naming is a question for @dwsilk maybe we could do `historic-survey-id`?

Reply:
Asked Daniel, makes sense to keep it as number despite it being a string input as it is called number everywhere else i.e., SN stands for `Survey Number`.

Unfortuantely we can't take a number and append `SN` as sometimes it might be `SNC` or have an `A` or `B` at the end.